auction-server/src/server.rs

```diff
@@ -93,11 +95,10 @@ use {
 async fn fault_tolerant_handler<F, Fut>(name: String, f: F)
 where
     F: Fn() -> Fut,
-    Fut: Future<Output = anyhow::Result<()>> + Send + 'static,
-    Fut::Output: Send + 'static,
+    Fut: Future<Output = anyhow::Result<()>>,
 {
     loop {
-        let res = tokio::spawn(f()).await;
+        let res = AssertUnwindSafe(f()).catch_unwind().await;
```

Comment:
This is a superior way of doing this; there's no reason to spawn a task that we await immediately.
Internally tokio uses AssertUnwindSafe too.

Reply:
Could you please describe what's the difference btw these two methods?